torch/utils/_sympy/reference.py

```diff
@@ -0,0 +1,122 @@
+import sympy
+
+# The normal Python interpretation of the operators
+# NB: For magic methods this needs to use normal magic methods
+# so that test_magic_methods works
+class ReferenceAnalysis:
+    @staticmethod
+    def constant(c, dtype):
```

Comment:
ignored dtype 🤔

Reply:
that's what ValueRangeAnalysis did too lol

Reply:
I actually don't really understand how this part is supposed to work, if someone could explain it to me that would be great.

Reply:
Passes in a constant value and an associated torch dtype: https://github.com/pytorch/pytorch/blob/master/torch/_inductor/lowering.py#L1518

Reply:
This seems... OK to ignore? Because we don't have any notion of size on the reference analysis. Unless I'm obligated to convert an int constant into float if the constant and dtype mismatch?

Reply:
Yes, i think we're fine to ignore it. Although it would be more safe to do the conversion in case the callee had a mismatch